src/Microsoft.ML.Onnx/OnnxUtils.cs

```diff
@@ -295,56 +295,40 @@ public ModelArgs(string name, TensorProto.Types.DataType dataType, List<long> di
             Contracts.CheckNonEmpty(colName, nameof(colName));
 
             TensorProto.Types.DataType dataType = TensorProto.Types.DataType.Undefined;
-            DataKind rawKind;
+            Type rawType;
             if (type is VectorType vectorType)
-                rawKind = vectorType.ItemType.RawKind;
-            else if (type is KeyType keyType)
-                rawKind = keyType.RawKind;
+                rawType = vectorType.ItemType.RawType;
+            else
+                rawType = type.RawType;
+
+            if (rawType == typeof(bool))
+                dataType = TensorProto.Types.DataType.Float;
```

Comment:
>Float [](start = 54, length = 5)

bool?

Reply:
I was wondering the same thing when I made the change...

Reply:
This probably is caused by the limited types accepted by old ONNX operators implemented in RS4. As ONNXRuntime supports much more ops than dark-age implementations, we should revise our type mappings and create more tests. This is a reasonable setting but it can be improved in another PR.

---
In reply to: [247951442](https://github.com/dotnet/machinelearning/pull/2143#discussion_r247951442) [](ancestors = 247951442)

Reply:
Looks like we have an existing bug to make this better - https://github.com/dotnet/machinelearning/issues/1198